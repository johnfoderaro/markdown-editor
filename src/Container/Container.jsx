import React from 'react';
import { hot } from 'react-hot-loader';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Block = styled.div`
  padding: .5rem calc(50% - 680px);
  margin-right: 1rem;
  margin-left: 1rem;
`;

const Container = ({ children }) => (
  <Block>
    {children}
  </Block>
);

Container.propTypes = {
  children: PropTypes.element.isRequired,
};

export default hot(module)(Container);
