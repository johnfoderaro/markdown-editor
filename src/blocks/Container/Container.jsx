import React from 'react';
import { hot } from 'react-hot-loader';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Block = styled.div`
  height: 100%;
  min-height: 100%;
  display: flex;
  flex-direction: row;
`;

const Container = ({ children }) => (
  <Block>
    {children}
  </Block>
);

Container.propTypes = {
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
};

export default hot(module)(Container);
