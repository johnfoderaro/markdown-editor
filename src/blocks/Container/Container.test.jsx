import React from 'react';
import { shallow } from 'enzyme';

import Container from './Container';

let tree;

beforeAll(() => {
  tree = shallow(
    <Container>
      <h1>Test</h1>
    </Container>,
  );
});

describe('<Container />', () => {
  it('should be defined', () => {
    expect(tree).toBeDefined();
  });
  it('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });
  it('should render a h1', () => {
    expect(tree.find('h1').text()).toEqual('Test');
  });
});
