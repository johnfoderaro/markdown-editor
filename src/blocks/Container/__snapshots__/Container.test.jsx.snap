// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Container>
    <h1>
      Test
    </h1>
  </Container>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <h1>
        Test
      </h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test",
      },
      "ref": null,
      "rendered": "Test",
      "type": "h1",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  padding: .5rem calc(50% - 680px);
  margin-right: 1rem;
  margin-left: 1rem;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <h1>
          Test
        </h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test",
        },
        "ref": null,
        "rendered": "Test",
        "type": "h1",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  padding: .5rem calc(50% - 680px);
  margin-right: 1rem;
  margin-left: 1rem;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
